ynr/apps/parties/importer.py

```diff
@@ -0,0 +1,324 @@
+                    )
+                except Party.DoesNotExist:
+                    print("Can't find the folllowing")
+                    print(other_party_name)
+                    continue
+                joint_id = make_joint_party_id(
+                    description.party.ec_id, other_party.ec_id
+                )
+                joint_party, created = Party.objects.update_or_create(
+                    ec_id=joint_id,
+                    defaults={
+                        "name": joint_party_name,
+                        "date_registered": description.date_description_approved,
+                    },
+                )
+                if created:
+                    self.collector.append(joint_party)
+
+
+class ECParty(dict):
+    """
+    A python representation of the party JSON from the EC API
+
+    """
+
+    def __init__(self, party_dict):
+        self.update(**party_dict)
+        self.model = None
+        self.created = False
+        required_fields = ("RegulatedEntityName", "ECRef", "RegisterName")
+        for field in required_fields:
+            if not field in self:
+                raise ValueError("{} missing".format(field))
+
+    def save(self):
+        self.model, self.created = Party.objects.update_or_create(
+            ec_id=self.ec_id,
+            defaults={
+                "name": self.cleaned_name,
+                "register": self.register,
+                "status": self.registration_status,
+                "date_registered": self.parse_date(self["ApprovedDate"]),
+                "date_deregistered": self.date_deregistered,
+            },
+        )
+
+        for description in self.get("PartyDescriptions", []):
+            text = " | ".join(
+                [
+                    d
+                    for d in (
+                        description["Description"],
+                        description.get("Translation", None),
+                    )
+                    if d
+                ]
+            )
+            PartyDescription.objects.update_or_create(
+                description=text,
+                party=self.model,
+                defaults={
+                    "date_description_approved": self.parse_date(
+                        description["DateDescriptionFirstApproved"]
+                    )
+                },
+            )
+
+        for emblem_dict in self.get("PartyEmblems", []):
+            emblem = ECEmblem(self.model, emblem_dict)
+            emblem.save()
+
+        return (self.model, self.created)
+
+    @property
+    def ec_id(self):
+        return self["ECRef"]
+
+    @property
+    def cleaned_name(self):
+        name = self["RegulatedEntityName"]
+
+        # Remove [De-registered] tag in name
+        # (the date is parsed in `date_deregistered`)
+        matcher = re.compile("[\[\(]de(-?)registered .*", flags=re.I)
+        name = matcher.split(name)[0].strip()
+
+        # Do some general cleaning
+        name = name.replace("  ", " ")
```

Comment:
What if we've got 3 whitespace characters here? It would be more flexible do something like `name = re.sub( '\s+', ' ', name ).strip()` or `name = ' '.join(name.split())` to replace N whitespace chars with one?